.github/workflows/cd-dev-deploy.yml

```
@@ -0,0 +1,52 @@
+  cancel-in-progress: true
+
+env:
+  APP_NAME: search-service
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  APP_PORT: 9005
+
+jobs:
+  deploy:
+    # ✅ CI 성공 + dev 브랜치에서만
+    if: >
```

actionlint (1.7.9)

if: condition "${{\n  github.event.workflow_run.conclusion == 'success' &&\n  github.event.workflow_run.head_branch == 'dev'\n}}\n" is always evaluated to true because extra characters are around ${{ }}

(if-cond)